prow/github/types.go

```diff
@@ -118,3 +118,21 @@ type IssueComment struct {
 	User    User   `json:"user,omitempty"`
 	HTMLURL string `json:"html_url,omitempty"`
 }
+
+type StatusEvent struct {
+	SHA         *string `json:"sha,omitempty"`
```

Comment:
I've been using raw values rather than pointers elsewhere in this file. For consistency's sake, can we keep it that way for now? I understand the reason to use pointers, but they shouldn't really be necessary for this.

Instead of `if se.State == nil || se.Context == nil {` you can do `if se.State == "" || se.Context == "" {` for the same effect, and you don't have to worry about nil dereferences. For instance, you later dereference `se.SHA` without checking that it's not nil. In the strange case that github gives us a payload without a sha, this code crashes the whole ship, whereas not using a pointer here just makes some later function return an error that we log out.